### PR TITLE
Various updates to the code generator

### PR DESCRIPTION
**Pull request by devoncarew**
Various updates to the code generator (a redux of https://github.com/dart-lang/intl_translation/pull/189):
-  switch to using the new function typedefs
- cleanups to our use of `ignore_for_file` analyzer directives
- various other minor changes to the generated code

Done as separate commits for easier review -

### CODE REVIEW

File: lib/generate_localized.dart [hunk 1]
```diff
@@ -174,75 +174,76 @@ class MessageGeneration {
...
       output.write("import '$file' ");
       if (useDeferredLoading) output.write('deferred ');
       output.write('as ${libraryName(locale)};\n');
     }
     output.write('\n');
-    output.write('typedef Future<dynamic> LibraryLoader();\n');
-    output.write('Map<String, LibraryLoader> _deferredLibraries = {\n');
+    output.writeln('typedef LibraryLoader = Future<dynamic> Function();');
+    output.writeln();
+    output.write('final Map<String, LibraryLoader> _deferredLibraries = {\n');
```
Comment (mosuem) [inline lib/generate_localized.dart:246]:
```suggestion
    output.writeln('final Map<String, LibraryLoader> _deferredLibraries = {');
```

File: lib/generate_localized.dart [hunk 2]
```diff
@@ -174,75 +174,76 @@ class MessageGeneration {
...
-    for (var locale in allLocales) {
+    var sortedLocales = allLocales.toList()..sort();
+    for (var locale in sortedLocales) {
       var baseFile = '${generatedFilePrefix}messages_$locale.dart';
       var file = importForGeneratedFile(baseFile);
       output.write("import '$file' ");
       if (useDeferredLoading) output.write('deferred ');
       output.write('as ${libraryName(locale)};\n');
     }
     output.write('\n');
```
Comment (mosuem) [inline lib/generate_localized.dart:243]:
```suggestion
    output.writeln();
```

File: lib/generate_localized.dart [hunk 3]
```diff
@@ -174,75 +174,76 @@ class MessageGeneration {
...
     clearOutput();
     output.write(localesPrologue);
-    for (var locale in allLocales) {
+    var sortedLocales = allLocales.toList()..sort();
+    for (var locale in sortedLocales) {
       var baseFile = '${generatedFilePrefix}messages_$locale.dart';
       var file = importForGeneratedFile(baseFile);
       output.write("import '$file' ");
       if (useDeferredLoading) output.write('deferred ');
       output.write('as ${libraryName(locale)};\n');
```
Comment (mosuem) [inline lib/generate_localized.dart:241]:
```suggestion
      output.writeln('as ${libraryName(locale)};');
```

**Review by mosuem — APPROVED** [review]

**Comment by devoncarew** [comment]
(merged changes from https://github.com/dart-lang/intl_translation/pull/197 here)

**Review by mosuem — APPROVED** [review]

**Review by mosuem — APPROVED** [review]

**Comment by devoncarew** [comment]
(@mosuem, fyi, I probably won't land this anytime in the next few weeks; if you do want to pull your `ignore_for_file: type=lint` change out separately, feel free).

**Pull request closed without merging**